modules/enterprise/server/server-metrics/src/main/java/org/rhq/server/metrics/aggregation/DataAggregator.java

```diff
@@ -351,8 +351,10 @@ public AggregationTask(Batch batch) {
         @Override
         public void run() {
             try {
+                permits.acquire();
                 run(batch);
             } catch (Exception e) {
+                permits.release();
```

Comment:
The FutureCallback only gets scheduled at the very end[1] of the AggregationTask. 
If an exception is raised before reaching that it would release only. If the code that schedules the FutureCallback raises an exception after scheduling it would release twice. 

I could make sure to only release the semaphore if the operation doesn't get scheduled.

[1] https://github.com/josejulio/rhq/blob/bugs/1288455/modules/enterprise/server/server-metrics/src/main/java/org/rhq/server/metrics/aggregation/DataAggregator.java#L131